### DISCUSSION TITLE
Why ForeignKey is declared as Optional but scalar values are not?

### DISCUSSION THREAD

**Discussion by mturilin** [collerek/ormar#631] [Q&A]
Hey folks,

Why ForeignKey is declared as Optional but scalar values are not?

Let's say I have a class:

```
class SignInRequest(BaseMeta):
    id = ormar.UUID(uuid_format="string")
    phone_number = ormar.Text(nullable=False)
    otp_code: str = ormar.Text(nullable=True)
    customer: Customer | None = ormar.ForeignKey(Customer)
```

Am I right that I need to declare otp_code as not optional but the customer as optional?
If yes, why?

—Mikhail

**Comment by SepehrBazyar** [comment]
Hi @mturilin 
Type Annotation values are used to validate model data through pydantic, and parameter values of each field are set on the database side.

In the example you wrote, you can also give the `nullable=True` argument to `ForeignKey` so that this item becomes optional on the SQL side, and it also better to set `opt_code: str | None = ...`.

And Finally, whether a field is optional or not depends on your view of the whole project and there is no right or wrong.
